compiler/rustc_passes/src/dead.rs

```diff
@@ -1013,17 +1012,11 @@ fn check_mod_deathness(tcx: TyCtxt<'_>, module: LocalModDefId) {
                         if let ShouldWarnAboutField::Yes(is_pos) =
                             visitor.should_warn_about_field(field)
                         {
-                            let level = tcx
-                                .lint_level_at_node(
-                                    if is_pos {
-                                        is_positional = true;
-                                        lint::builtin::UNUSED_TUPLE_STRUCT_FIELDS
-                                    } else {
-                                        lint::builtin::DEAD_CODE
-                                    },
-                                    hir_id,
-                                )
-                                .0;
+                            if is_pos {
+                                is_positional = true;
+                            }
```

Comment:
Isn't this just `is_positional = is_pos` or `is_positional |= is_pos`?

Reply:
Also can't we just do a single check is variant is tuple-like or not, instead of trying to move this out of the loop?..

Reply:
> Isn't this [...] `is_positional |= is_pos`?

Yeah, I thought about that but wanted to make the diff smaller / easier to see how it changed. Happy to update to this.

> do a single check is variant is tuple-like or not

That seems plausible, but I don't know my way around the compiler enough to do so. Got any hints?

> instead of trying to move this out of the loop

I thought you were suggesting that we *should* move it out of the loop — I must not understand what you mean.

Reply:
> I thought you were suggesting that we should move it out of the loop — I must not understand what you mean.

Right, I see why this is confusing. 

> "trying to move _this_ out of the loop"

_this_ refers to the `is_pos` which is moved into `is_positional`, which is outside of the loop.